tests/unit/components/reception/SharedTestsForImportAndScanIn.js

```diff
@@ -0,0 +1,166 @@
+        wrapper.vm.reception.fetchFunction = vi.fn()
+        vi.spyOn(Reception, 'createReceptionResource').mockImplementation(() => {})
+      })
+      it('shows error alert message', () => {
+        expect(mockShowAlert).toHaveBeenCalledWith('Failed fetch', 'danger')
+      })
+
+      it('does not call create resource functionality', async () => {
+        expect(Reception.createReceptionResource).not.toHaveBeenCalled()
+      })
+
+      it('does call scan in LabWhere functionality', () => {
+        expect(scanBarcodesInLabwhereLocation).not.toHaveBeenCalled()
+      })
+    })
+
+    describe('Fails to import labware', () => {
+      store.state.traction.messages = []
+
+      it('calls create resource functionality', async () => {
+        await wrapper.vm.importLabware()
+        expect(Reception.createReceptionResource).toHaveBeenCalled()
+      })
+
+      it('shows error alert message', async () => {
+        const message = 'The princess is in another castle'
+        vi.spyOn(Reception, 'createReceptionResource').mockRejectedValueOnce(new Error(message))
+        await wrapper.vm.importLabware()
+        expect(mockShowAlert).toHaveBeenCalledWith(new Error(message), 'danger')
+      })
+
+      it('does not call scanIn LabWhere functionality', () => {
+        expect(scanBarcodesInLabwhereLocation).not.toHaveBeenCalled()
+      })
+    })
+  })
+
+  describe('Successful import', () => {
+    describe('Successfully import and scan in to LabWhere ', () => {
+      let mockedCreateReception
+      let mockedCreateMessages
+      let tractionReceptionsCreate
+      let foundBarcodes
+      let attributes
+      let importedBarcodes
+
+      beforeEach(async () => {
+        vi.resetAllMocks()
+        vi.clearAllMocks()
+        store.state.traction.messages = []
+        mockedCreateReception = vi
+          .spyOn(Reception, 'createReceptionResource')
+          .mockImplementation(() => {})
+        mockedCreateMessages = vi.spyOn(Reception, 'createMessages').mockImplementation(() => {
+          return [{ text: '1 labware(s) successfully imported to Traction', type: 'success' }]
+        })
+
+        tractionReceptionsCreate = wrapper.vm.api.traction.receptions.create
+
+        foundBarcodes = new Set(['NT1'])
+        attributes = { source: 'traction-ui.sequencescape', request_attributes: [{}] }
+        wrapper.vm.labwareData = { foundBarcodes, attributes }
+        importedBarcodes = Array.from(foundBarcodes)
+
+        scanBarcodesInLabwhereLocation.mockResolvedValueOnce({
+          success: true,
+          message: 'Scan in 1 barcode(s) to location location1',
+        })
+
+        await wrapper.vm.importLabware()
+      })
+
+      it('calls created reception functionality ', async () => {
+        expect(mockedCreateReception).toBeCalledWith(
+          tractionReceptionsCreate,
+          foundBarcodes,
+          attributes,
+        )
+      })
+
+      it('create response messages accordingly ', async () => {
```

Comment:
Having a test just to say something is called is probably not necessary. As long as the result of that call is being tested elsewhere.